### DISCUSSION TITLE
CLI command to actually pull artifacts without installing them

### DISCUSSION THREAD

**Discussion by boesing** [composer/composer#12173] [Q&A / Support] [answered]
Hey guys,

I wonder if there is a way to "warmup" the composer cache, i.e. with something similar to "docker pull".
My scenario would be:

I am in a `Dockerfile` and want to build a container for my PHP project.
When creating the container, I do want to have some composer dependencies pulled from packagist/private packagist so that these are available whenever I start the container and run composer install afterwards.

In development environment, we mount the whole project as a volume to the container and I do want to make use of the cache of the image which was build beforehand  in case there is no `vendor/` directory. I would take care of the cache directory of composer ofc but the current solution would be to copy `composer.json` and `composer.lock`, run `composer install` (which creates `vendor/`) and then delete `composer.*` files along with the `vendor/` directory again but we could basically safe some i/o if we skip the whole write/delete of `vendor/` directory.

Any thoughts on this?

**Comment by Seldaek** [comment, marked as answer]
You can do `composer install --download-only`

**Reply by boesing** [reply]
Oh, that is already an option? Nice!
